dashboard/src/components/AppView/AppControls/AppControls.tsx

```diff
@@ -30,21 +34,20 @@ class AppControls extends React.Component<IAppControlsProps, IAppControlsState>
   };
 
   public render() {
-    const { name, namespace } = this.props.app;
-    const deleted = this.props.app.info && this.props.app.info.deleted;
+    const { app, update } = this.props;
+    const { name, namespace } = app;
+    const deleted = app.info && app.info.deleted;
     if (!name || !namespace) {
       return <LoadingWrapper />;
     }
     return (
       <div className="AppControls">
         {/* If the app has been deleted hide the upgrade button */}
         {!deleted && (
-          <button className="button" onClick={this.handleUpgradeClick}>
-            Upgrade
-          </button>
-        )}
-        {this.state.upgrade && (
-          <Redirect push={true} to={`/apps/ns/${namespace}/upgrade/${name}`} />
+          <UpgradeButton
+            updateVersion={(update && update.latestVersion) || ""}
+            upgradeURL={`/apps/ns/${namespace}/upgrade/${name}`}
```

Comment:
this is a bit weird to me, why not just pass the namespace and name and let `UpgradeButton` construct the upgrade URL? Why would you need to use an UpgradeButton with a different upgrade URL?